### PR TITLE
Refactor Option: Remove autogenerated routers

### PR DESCRIPTION
**Pull request by joneubank**
This refactor removes the assignment of routes based on file path and undocumented file export requirements. This has several advantages:
- Available routes are defined in code not in file structure. This makes the paths explicitly declared instead of being built at run time. No more need for error handling cases where the file structure and content no longer matches the route import script.
- Ensures all routes are given the `BASE_ENDPOINT` prefix. No chance for errors by a new route not including this in their path. Routers can still be added in the `server.ts` file if they should not use the `BASE_ENDPOINT` prefix.
- Ensures no (unintentional) path collisions. The paths are all defined in the `routes/index.ts` file. This makes it easy to trace through the code to determine a route's path, and to see if you have any collisions since all the paths for a given router are handled in the same file.

In general I think this simplifies reading and maintaining the code by removing some of the magic that was happening.

### CODE REVIEW

File: src/constants/endpoint.ts [hunk 1]
```diff
@@ -1,9 +1,3 @@
-import urlJoin from 'url-join';
-
 export const BASE_ENDPOINT = '/api';
-export const JWKS_ENDPOINT = '/oauth/jwks';
 
-export const ARRANGER_READY_ENDPOINT = urlJoin(BASE_ENDPOINT, 'ping');
-export const AUTH_ENDPOINT = urlJoin(BASE_ENDPOINT, 'auth');
-export const HEALTH_ENDPOINT = urlJoin(BASE_ENDPOINT, 'health');
-export const STORAGE_ENDPOINT = urlJoin(BASE_ENDPOINT, 'storage');
+export const ARRANGER_READY_ENDPOINT = '/ping';
```
Comment (joneubank) [inline src/constants/endpoint.ts]: Note: This isn't used anywhere, should probably remove.

File: src/routes/health/health.router.ts [hunk 2]
```diff
@@ -1,26 +1,10 @@
...
-
-// getRoutes
-router.get('/routes', (req, res) => {
-  const routes = getRoutes();
-
-  res.status(200).send({
-    numberOfRoutes: routes.length,
-    routes: routes,
-  });
-});
```
Comment (joneubank) [inline src/routes/health/health.router.ts:26]: Refactor broke this so I removed it. We should provide a swagger for docs instead.

File: src/server.ts [hunk 3]
```diff
@@ -26,16 +26,7 @@ app.use(
...
-app.use(
-  morgan('dev', {
-    skip: (req, res) => {
-      // logs everything but health checks on dev, errors only otherwise
-      return process.env.NODE_ENV === 'development' || process.env.DEBUG === 'true'
-        ? [ARRANGER_READY_ENDPOINT, HEALTH_ENDPOINT].includes(req.originalUrl)
-        : res.statusCode < 400;
-    },
-  }),
-);
```
Comment (joneubank) [inline src/server.ts]: Its dev... just log everything :)

File: src/config/global.ts [hunk 4]
```diff
@@ -19,10 +19,10 @@
 
 import urlJoin from 'url-join';
 
-import { JWKS_ENDPOINT } from '../constants/endpoint';
 import logger from '../logger';
 import * as vault from './vault';
 
+const JWKS_URI_PATH = '/oauth/jwks';
```
Comment (justincorrigible) [inline src/config/global.ts:25]: 🎉

File: src/constants/endpoint.ts (hunk 1 above)
Comment (justincorrigible) [reply src/constants/endpoint.ts]: To document what we discussed over slack, this is used by K8s as arranger's own health endpoint.

https://github.com/supercluster-covid-data-portal/infra/blob/main/helm/api/dev/values.yaml#L64

File: src/routes/index.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+import AuthRouter from './auth/auth.router';
+import HealthRouter from './health/health.router';
+import StorageRouter from './storage/storage.router';
+
+const Routes = (): Router => {
+  const router = Router();
+
+  router.use('/auth', AuthRouter);
+  router.use('/health', HealthRouter);
```
Comment (justincorrigible) [inline src/routes/index.ts]: this route should be relocated to the root level (i.e. at `/`, not in `/api`), as this is meant to indicate the server's own health, regardless of the different endpoints' statuses.
we can update the value in the charts afterwards:
https://github.com/supercluster-covid-data-portal/infra/blob/main/helm/api/dev/values.yaml#L62

File: src/server.ts (hunk 3 above)
Comment (justincorrigible) [reply src/server.ts]: in this app, that is certainly not what you want to do... as logging everything would include an endless barrage of health pings in the logs.

For prod, I log only errors (400). no need to log every single successful request.
For dev however (or debugging, while in prod), I do log 200 etc. to assist troubleshooting, just avoiding the constant health pings.

**Review by justincorrigible — APPROVED** [review]
squash this puppy!

**Pull request merged**